cmake/Platform/Linux/Install_linux.cmake

```
@@ -6,4 +6,19 @@
 #
 #
 
-include(cmake/Platform/Common/Install_common.cmake)
\ No newline at end of file
+#! ly_install_code_function_override: Linux-specific copy function to handle RPATH fixes
+function(ly_install_code_function_override)
+    install(CODE
+"function(ly_copy source_file target_directory)
+    file(COPY \"\${source_file}\" DESTINATION \"\${target_directory}\" FILE_PERMISSIONS ${LY_COPY_PERMISSIONS} FOLLOW_SYMLINK_CHAIN)
+
+    get_filename_component(target_filename_ext \"\${source_file}\" LAST_EXT)
+    if(\"\${source_file}\" MATCHES \"qt/plugins\" AND \"\${target_filename_ext}\" STREQUAL \".so\")
+        get_filename_component(target_filename \"\${source_file}\" NAME)
+        file(RPATH_CHANGE FILE \"\${target_directory}/\${target_filename}\" OLD_RPATH \"\\$ORIGIN/../../lib\" NEW_RPATH \"\\$ORIGIN/..\")
+    endif()
+endfunction()"
+    )
```

This can be simplified a lot by using the CMake [bracket argument](https://cmake.org/cmake/help/v3.19/manual/cmake-language.7.html#bracket-argument) which acts like a raw string literal

---

So wrap the entire `function(ly_copy ...` string in `[[` and `]]`?

---

Here is how it can be done
```suggestion
[=[
"function(ly_copy source_file target_directory)
    file(COPY "${source_file}" DESTINATION "${target_directory}" FILE_PERMISSIONS ]=] "${LY_COPY_PERMISSIONS}" [=[ FOLLOW_SYMLINK_CHAIN)

    get_filename_component(target_filename_ext "${source_file}" LAST_EXT)
    if("${source_file}" MATCHES "qt/plugins" AND "${target_filename_ext}" STREQUAL ".so")
        get_filename_component(target_filename "${source_file}" NAME)
        file(RPATH_CHANGE FILE "${target_directory}/${target_filename}" OLD_RPATH "$ORIGIN/../../lib" NEW_RPATH "$ORIGIN/..")
    endif()
endfunction()"
    )
]=]
```

---

Alright, I'll give it a shot.

---

For the string configure approach this can be done.
```suggestion
set(ly_copy_template [=[
"function(ly_copy source_file target_directory)
    file(COPY "${source_file}" DESTINATION "${target_directory}" FILE_PERMISSIONS @LY_COPY_PERMISSIONS FOLLOW_SYMLINK_CHAIN)
    get_filename_component(target_filename_ext "${source_file}" LAST_EXT)
    if("${source_file}" MATCHES "qt/plugins" AND "${target_filename_ext}" STREQUAL ".so")
        get_filename_component(target_filename "${source_file}" NAME)
        file(RPATH_CHANGE FILE "${target_directory}/${target_filename}" OLD_RPATH "$ORIGIN/../../lib" NEW_RPATH "$ORIGIN/..")
    endif()
endfunction()"
    )
]=])
string(CONFIGURE "${ly_copy_template}" ly_copy_command @ONLY)
```